pkg/virtual/framework/types.go

```diff
@@ -45,6 +49,9 @@ type ReadyFunc func() error
 type VirtualWorkspace interface {
 	GetName() string
 	ResolveRootPath(urlPath string, context context.Context) (accepted bool, prefixToStrip string, completedContext context.Context)
+	// GetKubeContextPaths returns a map in which keys are the kubeconfig context names of published endpoints
+	// provided by this virtual workspace, and values are associated endpoint path prefix.
+	GetKubeContextPaths() map[string]string
```

Comment:
Seems like a weird factoring - why does the server have an opinion on what some client calls these endpoints in their local kubeconfig?

Reply:
It's not on their local kubeconfig. These are the contexts that the virtual workspace wants to be published in the kubeconfig that will be created by the virtual workspaces Root API Server command.

Reply:
I understand. Why does the _server_ care about kubeconfig contexts? These are separate, unrelated domains.